samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs

```diff
@@ -0,0 +1,664 @@
+
+            // Initialize helper class used to render the skill results on screen
+            m_renderer = new ObjectTrackRenderer(UIOverlayCanvas);
+
+            m_lock.Wait();
+            {
+                NotifyUser("Initializing skills...");
+                m_detectorDescriptor = new ObjectDetectorDescriptor();
+                m_detectorExecutionDevices = await m_detectorDescriptor.GetSupportedExecutionDevicesAsync();
+                await InitializeObjectDetectorAsync();
+
+                m_trackerDescriptor = new ObjectTrackerDescriptor();
+                m_trackerExecutionDevices = await m_trackerDescriptor.GetSupportedExecutionDevicesAsync();
+                await InitializeObjectTrackerAsync();
+
+                await UpdateSkillUIAsync();
+            }
+            m_lock.Release();
+
+            // Ready to begin, enable buttons
+            NotifyUser("Skill initialized. Select a media source from the top to begin.");
+            await UpdateMediaSourceButtonsAsync(true);
+        }
+
+        /// <summary>
+        /// Populate UI with skill information and options
+        /// </summary>
+        /// <returns></returns>
+        private async Task UpdateSkillUIAsync()
+        {
+            if (Dispatcher.HasThreadAccess)
+            {
+                // Show skill description members in UI
+                UISkillDescription.Text = m_detectorDescriptor.Information.Name;
+                UISkillDescription.Text += "\n" + SkillHelper.SkillHelperMethods.GetSkillDescriptorString(m_detectorDescriptor);
+                UISkillDescription.Text += "\n\n" + m_trackerDescriptor.Information.Name;
+                UISkillDescription.Text += "\n" + SkillHelper.SkillHelperMethods.GetSkillDescriptorString(m_trackerDescriptor);
+
+                if (m_detectorExecutionDevices.Count == 0 || m_trackerExecutionDevices.Count == 0)
+                {
+                    NotifyUser("No execution devices available, one or more skills cannot run on this device");
+                }
+                else
+                {
+                    // Display available detector execution devices
+                    UIDetectorExecutionDevices.ItemsSource = m_detectorExecutionDevices.Select((device) => $"{device.ExecutionDeviceKind} | {device.Name}");
+                    
+                    // Set SelectedIndex to index of currently selected device
+                    for (int i = 0; i < m_detectorExecutionDevices.Count; i++)
+                    {
+                        if (m_detectorExecutionDevices[i].ExecutionDeviceKind == m_detectorBinding.Device.ExecutionDeviceKind
+                            && m_detectorExecutionDevices[i].Name == m_detectorBinding.Device.Name)
+                        {
+                            UIDetectorExecutionDevices.SelectedIndex = i;
+                            break;
+                        }
+                    }
+
+                    // Display available detector execution devices
```

Comment:
>detector [](start = 41, length = 8)

tracker

Reply:
nit: still there :p

---
In reply to: [335700595](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/60#discussion_r335700595) [](ancestors = 335700595)

Reply:
Shouldn't be, are you checking latest?